airflow-core/src/airflow/api_fastapi/core_api/routes/public/assets.py

```diff
@@ -318,8 +325,19 @@ def materialize_asset(
             f"More than one DAG materializes asset with ID: {asset_id}",
         )
 
-    dag: DAG | None
-    if not (dag := request.app.state.dag_bag.get_dag(dag_id)):
+    try:
+        dag: DAG = request.app.state.dag_bag.get_dag(dag_id)
+    except (ImportError, SyntaxError):
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=f"Failed to parse DAG '{dag_id}'. Check DAG file syntax or dependencies.",
+        )
+    except Exception:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"An unexpected error occurred while trying to load DAG '{dag_id}'.",
+        )
```

Comment:
This is wrong. `DagBag.get_dag` returns `None` when there is an error. 

dag_bag will only have Dags that were successfully parsed.